portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -112,15 +124,72 @@ function insertSearch() {
 
   const searchInput = document.createElement('input');
   searchInput.className = 'form-control form-inline';
+  searchInput.id = 'userSearch';
   searchInput.type = 'text';
   searchInput.placeholder = 'Search';
 
   const searchI = document.createElement('i');
   searchI.className = 'fas fa-search text-white ml-3 mr-auto';
 
+  const searchResults = document.createElement('ul');
+  searchResults.id = 'searchResults';
+
   searchDiv.appendChild(searchInput);
   searchDiv.appendChild(searchI);
   searchBar.appendChild(searchDiv);
+  searchBar.append(searchResults);
+
+  return searchBar;
+}
+
+// searches each child Node of the page in the elementArray and retains the
+// elements that contain the wanted word
+function searchPages(elementArray) {
+  const wantedWords = document.getElementById('userSearch').value.toLowerCase();
+  const arrayLength = elementArray.length;
+
+  const resultArray = [];
+  const backArray = [];
+  if (wantedWords.length > 0) {
+    for (let i = 0; i < arrayLength / 2; i++) {
+      const frontElement = elementArray[i];
+      let frontElementText = frontElement.innerText;
+      if (frontElementText) {
+        frontElementText = frontElementText.toLowerCase();
+        if (frontElementText.includes(wantedWords)) {
+          resultArray.push(frontElement);
+        }
+      }
+
+      const backElement = elementArray[arrayLength - 1 - i];
+      let backElementText = backElement.innerText;
+      if (backElementText) {
+        backElementText = backElementText.toLowerCase();
+        if (backElementText.includes(wantedWords)) {
+          backArray.unshift(backElement);
+        }
+      }
+    }
+  }
 
-  document.getElementById('mainNav').appendChild(searchBar);
+  resultArray.concat(backArray);
+  return resultArray;
+}
+
+// populates the search skeleton with the results and sets the functionality to
+// navigate to the result by clicking on it.
+function showResults(resultArray) {
+  const searchResults = document.getElementById('searchResults');
+  searchResults.innerHTML = '';
+
+  resultArray.forEach(function(result) {
+    const textElement = document.createElement('li');
+    textElement.innerText = result.innerText.substring(0, 10);
+
+    textElement.onclick = function() {
+      result.scrollIntoView();
```

Comment:
@wangtong-g comment
https://github.com/kofimeighan/adelardos-capstone-2020/pull/17#discussion_r450559729
Could you turn the forEach inner function into a lambda?

Reply:
Yeah sure Tong, I will change this function.